### PR TITLE
Separate unit tests and integration tests

### LINKED ISSUE #1391: Separate unit tests and DB/integration tests
## Summary
Currently all tests are in one group. It would be better to keep them separate.

## Motivation
Unit tests should be blazingly fast. The DB tests are taking long and slowing down the entire CI.

## Solution
- Split the two types of tests into separate bundles/groups
- Move DB tests to nightly on CI

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #1391

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,8 @@ project(clio VERSION ${CLIO_VERSION} HOMEPAGE_URL "https://github.com/XRPLF/clio
 
 # =========================== Options ====================================== #
 option(verbose "Verbose build" FALSE)
-option(tests "Build tests" FALSE)
+option(tests "Build unit tests" FALSE)
+option(integration_tests "Build database tests" FALSE)
```
Comment (kuznetsss) [inline CMakeLists.txt]: I think here and in conanfile these should be `true` by default.

File: tests/unit/CMakeLists.txt [hunk 2]
```diff
@@ -174,7 +164,7 @@ if (coverage)
     --gtest_brief=1
     ${TESTS_ADDITIONAL_ARGS}
     EXCLUDE
-    "unittests"
+    "tests"
```
Comment (kuznetsss) [inline tests/unit/CMakeLists.txt:167]: You need to exclude here all the backend part you excluded in CI.

File: CMakeLists.txt (hunk 1 above)
Comment (godexsoft) [reply CMakeLists.txt]: I don't think so.. why would it be? 
Normal use-case for non-clio-devs is to build Clio for running it.. why would they need to mess around with unit and integration tests? It was false all along for this reason.

**Review by cindyyan317 — APPROVED** [review]
LGTM

**Pull request merged**